code/parachain/frame/staking-rewards/src/test/mod.rs

```diff
@@ -85,6 +85,81 @@ fn test_create_reward_pool_invalid_end_block() {
 	});
 }
 
+#[test]
+fn create_staking_reward_pool_should_fail_when_pool_asset_id_is_zero() {
+	new_test_ext().execute_with(|| {
+		System::set_block_number(1);
+
+		assert_err!(
+			StakingRewards::create_reward_pool(
+				Origin::root(),
+				RewardRateBasedIncentive {
+					owner: ALICE,
+					asset_id: 0,
+					// end block can't be before the current block
+					start_block: 2,
+					end_block: 0,
+					reward_configs: default_reward_config(),
+					lock: default_lock_config(),
+					share_asset_id: XPICA::ID,
+					financial_nft_asset_id: STAKING_FNFT_COLLECTION_ID,
+				}
+			),
+			crate::Error::<Test>::InvalidAssetId,
+		);
+	});
+}
+
+#[test]
+fn create_staking_reward_pool_should_fail_when_share_asset_id_is_zero() {
+	new_test_ext().execute_with(|| {
+		System::set_block_number(1);
+
+		assert_err!(
+			StakingRewards::create_reward_pool(
```

Comment:
typical example where types would no need to write test. either use Validated OR create AssetId type with one line to serde back and check it is non null from some NonZero128. it would take several times less lines but universally reusable. and also reusable in arbitrary becase it will never generate 1. cc @benluelo

Reply:
Yep, agreed - we should be using `Validated` here

Reply:
After talking to the rest of the Parachain finance team this morning, we've decided not to move this to Validated yet.
Currently, Validated does not properly emit errors to the runtime due to an oversight in its design.
Eventually, it would be good to move this, and other checks, to validated but we can't be doing that until we are receiving accurate errors.